### ISSUE TITLE
union: --mirror option for union backend

### ISSUE THREAD

**Issue by tsumare** [rclone/rclone#3308]
#### What is your current rclone version (output from `rclone version`)?

```
rclone v1.48.0
- os/arch: linux/amd64
- go version: go1.12.6
```

#### What problem are you are trying to solve?

Keep multiple remote (backup) targets in sync

#### How do you think rclone should be changed to solve that?

Add a `mirror` option for the union backend which causes it to write any new or changed files to all listed backends rather than simply the last one.  This would allow the use of a single rclone target which puts my backup files on google drive, dropbox, all at once.  This mode of operation would allow users to configure the union backend to be "read top, write all", rather than "read top, write top".

**Comment by ncw** [comment]
That sounds like a good feature.  It wouldn't be too hard to implement I don't think.

Do you want to have a go?

**Comment by gary-kim** [comment]
I can work on this. It'd probably be better to make a `--union-mirror` option.

**Comment by ncw** [comment]
> I can work on this. It'd probably be better to make a `--union-mirror` option.

Great :-)

> EDIT: Also, how should errors be handled? My current implementation does the operation on all remotes then reports the error of the last remote to return an error or nil if none returned an error. Is there a better way this could be done?

That seems like a reasonable thing to do.  However it does mean retries are going to be inefficient as it will potentially upload stuff to all remotes all over again just because one failed.

**Comment by J0s3f** [comment]
Just to add one usecase I'd really like that should be possible with that solution:

I want to have all files mirrored on two different cloud providers. I have slow upload at home but access to an server with fast upload.

What I would like is:
Setup both cloud providers and a mirror config on my server **S**.
Export the mirror from **S** using encrypted WebDAV with rclone.
Mount the WebDAV form **S** on my PC **P**.

If I add a new file, it should be uploaded form **P** to **S** only once, needing only the size of the file plus overhead as traffic.
From **S**, the file should be sent to both cloud providers.

I think if there is no retries for failures, a copy should only be successful if the file is stored at all configured mirrors.

A better option would be to be able to configure number of copies a file needs for success and then automatically retry failed providers later.
E.g. configure 1 copy, so the server returns OK if the file is at one provider. But it then will automatically try to copy the file from the one provider to the other later.
But this option is probably harder to implement and leaves open the question how permanent failures should be handled/the user notified.

**Comment by ivandeex** [comment]
Mirroring (as well as more advanced policies) have been implemented by https://github.com/rclone/rclone/pull/3782 (multi-write union).
I suggest closing this issue.

Please refer to new _union_ manual https://rclone.org/union/
@J0s3f if your use case is not covered by create/update policies, I suggest you open a new issue.

**Comment by ivandeex** [comment]
Closing after 2 months of inactivity

> Mirroring (as well as more advanced policies) have been implemented by #3782 (multi-write union).
> I suggest closing this issue.
